### DISCUSSION TITLE
FlexBox woes ... number of columns based on screen width

### DISCUSSION THREAD

**Discussion by drmancini** [quasarframework/quasar#15660] [General - Components / Directives / etc]
I'm trying to create a layout, where I have 12 divs placed inside one parent div, and I am trying to display these divs in multiple columns in the following way:

**Wide screen**
Column 1: div 1, 2, 3, 4
Column 2: div 5, 6, 7, 8
Column 3: div 9, 10, 11, 12

**Narrow screen**
Column 1: div 1, 2, 3, 4, 5, 6
Column 2: div 7, 8, 9, 10, 11, 12

**Mobile**
Column 1: div 1, 2, 3, 4, 5, 6

I think I have tried many combinations of various classes but just can't get this to work. I understand how FlexBox works in rows but this just eludes me.

**Comment by suleimanas** [comment]
https://quasar.dev/layout/grid/row#responsive-classes
```
  <div class="row">
    <div class="col-2 col-lg-3" v-for="i in 12">{{i}}</div>
  </div>
```

**Reply by drmancini** [reply]
Thank you but this doesn't solve the issue. When the parent `class="row"` the child elements are rendered from left to right and when they reach the right edge they start a new line and continue.

This way with 3 columns you end up with column 1 being: div1, div 4, div 7. But I need the divs to wrap in columns and have column 1 containing div1, div2, div3.

**Reply by suleimanas** [reply]
sorry :) you keep saying column and I keep seeing row. my mistake

**Comment by pdanpdan** [comment]
I cannot see a solution for what you want in CSS

A solution with mostly JS:
https://codepen.io/pdanpdan/pen/poOXpLo?editors=1010

The only way with little JS is:
- calculate the column heights (number of items)
- create a `.row` with as many `.column.col-[4|6|12]` as you need
- fill the cells in the column 1 from 0 to col1 limit, in column 2 from col1 limit + 1 to col1 limit + col2 limit, ...
